### PR TITLE
Improve container base image detection

### PR DESCRIPTION
**Pull request by davidfowl**
- Improve ASP.NET project detection to work with netcoreapp2.1 applications
- Grab the target framework name and version from the project separately so we can use the vesion to determine the base image
- Updated Dockerfile generation and tye run --docker to work with different container versions based on the TFM version.

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 1]
```diff
@@ -83,10 +83,9 @@ private async Task TransformProjectToContainer(Service service, ProjectRunInfo p
             serviceDescription.RunInfo = dockerRunInfo;
         }
 
-        private static string DetermineContainerImage(string targetFramework)
+        private static string DetermineContainerImage(ProjectRunInfo project)
         {
-            // TODO: Determine the base image from the tfm
-            return "mcr.microsoft.com/dotnet/core/sdk:3.1-buster";
+            return $"mcr.microsoft.com/dotnet/core/sdk:{project.TargetFrameworkVersion}";
```
Comment (spboyer) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: This pattern would assume that all development, if using Linux containers, would be done on Debian based containers given the TFM base image is always such. How would we specify an alpine or other flavor? Should we consider looking leveraging  [Container MS Build Properties](https://docs.microsoft.com/en-us/visualstudio/containers/container-msbuild-properties?view=vs-2019) like `DockerfileTag` here?

Comment (davidfowl) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: Not possible to override it today. I’d imagine we’ll eventually let you control the base image or docker file completely

Comment (davidfowl) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: Seems reasonable to support these in msbuild for alignment anyways.

Comment (davidfowl) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: @spboyer can you file an issue for this?

Comment (spboyer) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: Done - https://github.com/dotnet/tye/issues/240

File: src/Microsoft.Tye.Core/ProjectReader.cs [hunk 2]
```diff
@@ -219,10 +219,18 @@ private static void EvaluateProject(OutputContext output, ProjectServiceBuilder
             project.TargetFramework = targetFramework;
             output.WriteDebugLine($"Found target framework: {targetFramework}");
 
+            // TODO: Parse the name and version manually out of the TargetFramework field if it's non-null
+            project.TargetFrameworkName = projectInstance.GetPropertyValue("_ShortFrameworkIdentifier");
+            project.TargetFrameworkVersion = projectInstance.GetPropertyValue("_ShortFrameworkVersion") ?? projectInstance.GetPropertyValue("_TargetFrameworkVersionWithoutV");
+
```
Comment (rynowak) [inline src/Microsoft.Tye.Core/ProjectReader.cs]: This really seems suspect :( do we know that 3.1 and 5.0 SDKS behavre the same way?

File: src/Microsoft.Tye.Hosting/DockerRunner.cs [hunk 3]
```diff
@@ -99,7 +99,8 @@ async Task RunDockerContainer(IEnumerable<(int ExternalPort, int Port, int? Cont
                     // Default to development environment
                     ["DOTNET_ENVIRONMENT"] = "Development",
                     // Remove the color codes from the console output
-                    ["DOTNET_LOGGING__CONSOLE__DISABLECOLORS"] = "true"
+                    ["DOTNET_LOGGING__CONSOLE__DISABLECOLORS"] = "true",
+                    ["ASPNETCORE_LOGGING__CONSOLE__DISABLECOLORS"] = "true"
```
Comment (rynowak) [inline src/Microsoft.Tye.Hosting/DockerRunner.cs]: OOOOH yeah!

File: src/Microsoft.Tye.Core/ProjectReader.cs (hunk 2 above)
Comment (davidfowl) [reply src/Microsoft.Tye.Core/ProjectReader.cs]: Yes, they do, at least by default they do. I still think we should do the janky fallback to parsing the tfm.

**Pull request merged**